teps/0070-tekton-catalog-task-platform-support.md

```diff
@@ -0,0 +1,137 @@
+
+### Goals
+
+- Define the way to add supported platforms into resource definition in the Tekton catalog.
+- Define the ways to show information about platforms to the user.
+
+### Non-Goals
+
+- Define modifications in Tekton to better support running tasks on different platforms ([TEP-0041](https://github.com/tektoncd/community/pull/310)).
+- Limit/hide ("no-show" or "no-access") catalog resources based on host platform.
+
+### Use Cases (optional)
+
+- Filter the resources in the Hub UI using platforms.
+- Search the resources with `tkn hub` for specific platform(s).
+- Get information about supported platforms for specific resource in the Hub UI/`tkn hub`.
+
+## Requirements
+
+- Platform information in the catalog resource should be optional.
+- Information how to identify/prove supported platform for the resource should be documented in the Tekton catalog.
+- Tekton Hub should automatically get/update information about platforms of the resources.
+- Hub UI should show all platforms available in database to filter.
+
+## Proposal
+
+### Catalog side
+
+Add new `tekton.dev/platforms` annotation to the resource yaml file with list of platforms separated by comma, for example
+
+```
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: image-build
+    tekton.dev/platforms: linux/amd64,linux/s390x
+```
+
+The `tekton.dev/platforms` annotation should be optional. Best practice is to provide information about supported platforms, which should be documented in the catalog README.
+If the annotation is specified, it should not be extended, modified at the processing steps. For instance, if only `linux/ppc64le` value is provided, `linux/amd64` should not be meant as also supported. 
+
+In case if the annotation is not specified, `linux/amd64` platform is meant by default, as this platform is actually default one in cloud/container world.
+This statement also provides backword compatibility to the old versions of the tasks in catalog.
+
+Information about supported platforms also should be added to the `README.md` as separate `Platform` section with extra use case examples for specific platforms if it is required.
+For instance, another value for default image parameter in the task.
+
+The conditions to add platforms into `tekton.dev/platforms` should be written in Tekton catalog documentation.
+The levels of verification are:
+- all the container images, used in the resource, have corresponding platform support.
+- the resource tests were executed on corresponding platform once for concrete version.
+- the resource tests are periodically executed on corresponding platform.
+
+The tests are usually already available in the same Tekton catalog directory as task resource itself.
+
+### Hub side
```

Comment:
Are the Hub changes meant to be in scope for this TEP?
There might be changes needed to the HUB API too.